### PR TITLE
PMM-6038 PT summary.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

File: actions/pt_summary_action.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	buf := new(bytes.Buffer)
+	cmd := exec.CommandContext(ctx, "pt-summary")
+	cmd.Dir = cfg.Paths.PTSummary
+	cmd.Stdout = buf
+	cmd.Stderr = new(bytes.Buffer)
+	if err := cmd.Run(); err != nil {
```
Comment (BupycHuk) [inline actions/pt_summary_action.go:62]: we should pass
```
	pdeathsig.Set(cmd, unix.SIGKILL)
```

File: actions/pt_summary_action.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return nil, err
```
Comment (BupycHuk) [inline actions/pt_summary_action.go:63]: Should we return nil as result? Is it possible to have non-zero exit code but usefull output?

File: actions/pt_summary_action.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// Run runs an Action and returns output and error.
+func (a *ptSummaryAction) Run(ctx context.Context) ([]byte, error) {
+	l := logrus.WithField("component", "pt-summary")
+	cfg, _, err := config.Get(l)
+	if err != nil {
+		return nil, err
+	}
+
+	buf := new(bytes.Buffer)
+	cmd := exec.CommandContext(ctx, "pt-summary")
```
Comment (BupycHuk) [inline actions/pt_summary_action.go:58]: Maybe it's better to extract logic related to starting, stopping and getting logs into separate structure? We will have other similar actions.

File: actions/pt_summary_action_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	t.Skip()
+
+	a := NewPTSummaryAction("")
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	res, err := a.Run(ctx)
+	require.NoError(t, err)
+	require.NotEmpty(t, res)
+	require.Regexp(t, regexp.MustCompile("# Percona Toolkit System Summary Report ######################"), string(res))
```
Comment (BupycHuk) [inline actions/pt_summary_action_test.go:37]: We can use strings.contains instead of regexp.

File: config/config.go [hunk 5]
```diff
@@ -89,7 +89,8 @@ type Paths struct {
 	ProxySQLExporter string `yaml:"proxysql_exporter"`
 	RDSExporter      string `yaml:"rds_exporter"`
 
-	TempDir string `yaml:"tempdir"`
+	TempDir   string `yaml:"tempdir"`
+	PTSummary string `yaml:"pt_summary"`
```
Comment (BupycHuk) [inline config/config.go:93]: let's keep it in a separate block

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: config/config_test.go [hunk 6]
```diff
@@ -84,6 +84,7 @@ func TestGet(t *testing.T) {
 		actual, configFilepath, err := get([]string{
 			"--id=agent-id",
 			"--server-address=127.0.0.1",
+			"--paths-pt-summary=/usr/local/percona/pmm2/pt-summary",
```
Comment (BupycHuk) [inline config/config_test.go:87]: let's remove it and check default value. To learn where it placed ask @EvgeniyPatlan

File: config/config_test.go [hunk 7]
```diff
@@ -118,6 +120,9 @@ func TestGet(t *testing.T) {
 			Server: Server{
 				Address: "127.0.0.1",
 			},
+			Paths: Paths{
```
Comment (BupycHuk) [inline config/config_test.go:123]: Let's remove it too, we should have default value

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: client/client.go [hunk 8]
```diff
@@ -300,6 +300,9 @@ func (c *Client) processChannelRequests() {
 			case *agentpb.StartActionRequest_MongodbQueryGetcmdlineoptsParams:
 				action = actions.NewMongoDBQueryAdmincommandAction(p.ActionId, params.MongodbQueryGetcmdlineoptsParams.Dsn, "getCmdLineOpts", 1)
 
+			case *agentpb.StartActionRequest_PtSummaryParams:
+				action = actions.NewProcessAction(p.ActionId, c.cfg.Paths.PTSummary, []string{})
```
Comment (BupycHuk) [inline client/client.go:304]:
```suggestion
				action = actions.NewProcessAction(p.ActionId, c.cfg.Paths.PTSummary, new([]string))
```

**Review by BupycHuk — APPROVED** [review]
Only one minor comment

File: client/client.go (hunk 8 above)
Comment (AlekSi) [reply client/client.go]: That will not compile

Comment (JiriCtvrtka) [reply client/client.go:304]: Yes, I already discard it. Thanks.

**Comment by JiriCtvrtka** [comment]
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1065

**Pull request merged**